Tests/Unit/Office365DSC/Office365DSC.O365AdminAuditLogConfig.Tests.ps1

```diff
@@ -0,0 +1,180 @@
+                (Get-TargetResource @testParams).UnifiedAuditLogIngestionEnabled | Should Be 'Enabled'
+            }
+
+            It 'Should return false from the Test method' {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It 'Enables UnifiedAuditLogIngestionEnabled in the Set method' {
+                Set-TargetResource @testParams
+            }
+        }
+
+
+        Context -Name 'Test Passes When the Unified Audit Log Ingestion is Disabled' -Fixture {
+            $testParams = @{
+                IsSingleInstance                = 'Yes'
+                Ensure                          = 'Present'
+                GlobalAdminAccount              = $GlobalAdminAccount
+                UnifiedAuditLogIngestionEnabled = 'Disabled'
+            }
+
+            Mock -CommandName Get-AdminAuditLogConfig -MockWith {
+                return @{
+                    UnifiedAuditLogIngestionEnabled = $false
+                }
+            }
+
+            It 'Should return Disabled from the Get method' {
+                (Get-TargetResource @testParams).UnifiedAuditLogIngestionEnabled | Should Be 'Disabled'
+            }
+
+            It 'Should return true from the Test method' {
+                Test-TargetResource @testParams | Should Be $true
+            }
+
+        }
+
+        Context -Name 'Test Passes When the Unified Audit Log Ingestion is Enabled' -Fixture {
+            $testParams = @{
+                IsSingleInstance                = 'Yes'
+                Ensure                          = 'Present'
+                GlobalAdminAccount              = $GlobalAdminAccount
+                UnifiedAuditLogIngestionEnabled = 'Enabled'
+            }
+
+            Mock -CommandName Get-AdminAuditLogConfig -MockWith {
+                return @{
+                    UnifiedAuditLogIngestionEnabled = $true
+                }
+            }
+
+            It 'Should return Enabled from the Get method' {
+                (Get-TargetResource @testParams).UnifiedAuditLogIngestionEnabled | Should Be 'Enabled'
+            }
+
+            It 'Should return true from the Test method' {
+                Test-TargetResource @testParams | Should Be $true
+            }
+
+        }
+
+        Context -Name 'Test Fails When the Ensure Absent is specified' -Fixture {
+            $testParams = @{
+                IsSingleInstance                = 'Yes'
+                Ensure                          = 'Absent'
+                GlobalAdminAccount              = $GlobalAdminAccount
+                UnifiedAuditLogIngestionEnabled = 'Enabled'
+            }
+
+            Mock -CommandName Get-AdminAuditLogConfig -MockWith {
+                return @{
+                    UnifiedAuditLogIngestionEnabled = $true
+                }
+            }
+
+            It 'Should throw error from the Get method' {
+                Get-TargetResource @testParams | Should -Throw
```

Comment:
Use a syntax like the following instead:
{ Get-TargetResource @testParams } | Should Throw "This resource cannot delete Managed Properties. Please make sure you set its Ensure value to Present."

Reply:
Ok, I'll try that next